research/gnn-survey/layers.py

```diff
@@ -117,3 +117,73 @@ def call(self, inputs):
     attn = self.dropout(attn)
     output = tf.matmul(attn, x)
     return output
+
+
+class SpGraphAttnLayer(tf.keras.layers.Layer):
+  """ Single sparse graph attention layer."""
+
+  def __init__(self, output_dim, dropout_rate, alpha=0.2, concat=True, **kwargs):
```

Comment:
I don't understand the `concat=True` argument. It's not documented and looks like it actually refers to the elu activation?

Reply:
It should be removed, thanks:)